source/wp-content/plugins/wporg-internal-notes/src/sidebar.js

```diff
@@ -27,9 +29,34 @@ const SidebarIcon = () => {
 };
 
 export const NotesSidebar = () => {
-	const initialNotes = useSelect( ( select ) => {
-		return select( notesStore ).getNotes();
+	const { initialNotes, afterDate, isSaving } = useSelect( ( select ) => {
+		return {
+			initialNotes: select( notesStore ).getNotes(),
+			afterDate: select( notesStore ).getLatestNoteDate(),
+			isSaving: select( 'core/editor' ).isSavingPost(),
+		};
 	} );
+	const { prependNotes, clearIsCreated } = useDispatch( notesStore );
+	const prevIsSaving = usePrevious( isSaving );
+	const [ isLoading, setIsLoading ] = useState( false );
+
+	// Update the notes list when new notes are created, even if the sidebar is closed.
+	useEffect( () => {
+		// Gutenberg doesn't have a specific way to check whether the post was just saved, so we store the
+		// previous value of isSaving and when isSaving goes from true -> false, we prepend the notes.
+		const postWasJustSaved = prevIsSaving && ! isSaving;
+		if ( postWasJustSaved ) {
```

Comment:
It's important to note that this doesn't make any assumptions as to whether it succeeded. I'm not familiar with all the use cases of this code/plugin, but if you wanted to make sure it succeeded first, check:

```
select('core/editor').didPostSaveRequestSucceed()
```

This check **does not** represent the "next" state after saving so it needs to be checked _after_ a save was performed. For example, if you check this on load, it may return `true`.

Reply:
That's a good point. In this case, though, I don't think it matters whether or not it succeeded. If the routine to prepend more log notes runs, but doesn't find any new logs, it should just fetch an empty array, rather than throwing an error of some kind.